src/components/bridge-blocker.ts

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as logging from "./logging";
+const log = logging.get("BridgeBlocker");
+
+export class BridgeBlocker {
+    _isBlocked = false;
+
+    get isBlocked(): boolean {
+        return this._isBlocked;
+    }
+
+    constructor(private userLimit: number) {}
+
+    public checkLimits(users: number) {
+        log.debug(`Bridge now serving ${users} users`);
+
+        if (users > this.userLimit) {
+            if (!this._isBlocked) {
+                this.blockBridge().then(() => {
```

Comment:
Do we want to be nice here and catch failures? (I'd vote yes)

if not, we should warn implementers to catch their own failures.

Reply:
We're being nice now in 2a8861a7b640a61543e3481b25e09e4b53ca9b3c :)